File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> sorted = nums;
        sort(sorted.begin(), sorted.end());

        unordered_map<int, int> map;

        for (int i = 0; i < sorted.size(); i++) {
            if (map.find(sorted[i]) == map.end()) {
                map[sorted[i]] = i;
            }
        }

        vector<int> ans;

        for (int num : nums) {
            ans.push_back(map[num]);
        }

        return ans;
    }
};

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        sorted_nums = sorted(nums)

        first_index = {}

        for i, num in enumerate(sorted_nums):
            if num not in first_index:
                first_index[num] = i

        ans = []

        for num in nums:
            ans.append(first_index[num])

        return ans